src/logprovider/logprovider.hpp

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef LOGPROVIDER_HPP_
+#define LOGPROVIDER_HPP_
+
+#include <condition_variable>
+#include <mutex>
+#include <queue>
+#include <string>
+#include <thread>
+#include <vector>
+
+#include "archivator.hpp"
+#include "journal.hpp"
+
+namespace aos::sm::logprovider {
+
+/*
+ * Provides service instances ID.
+ */
+class InstanceIDProviderItf {
+public:
+    /**
+     * Returns service instance IDs.
+     *
+     * @param filter service instance filter.
+     * @return RetWithError<std::vector<std::string>>.
+     */
+    virtual RetWithError<std::vector<std::string>> GetInstanceIDs(const cloudprotocol::InstanceFilter& filter) = 0;
+
+    /**
+     * Destructor.
+     */
+    virtual ~InstanceIDProviderItf() = default;
+};
+
+/**
+ * Private struct for GetLog request.
+ */
+struct GetLogRequest {
```

Comment:
Looks like pure private struct for `LogProvider`. Move to LogProvider private section.

Reply:
yes, it is used inside LogProvider only,
but decided to move out of class declaration because private section looks too overloaded